src/library/searchquery.cpp

```diff
@@ -166,6 +172,46 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+CrateFilterNode::CrateFilterNode(const QSqlDatabase& database,
+                                 const QString& argument)
+    : m_database(database),
+      m_argument(argument) {
+    getTrackIds();
+}
+
+void CrateFilterNode::getTrackIds() {
+    FieldEscaper escaper(m_database);
+    QString escapedArgument = escaper.escapeString(kSqlLikeMatchAll + m_argument + kSqlLikeMatchAll);
+
+    QSqlQuery query(m_database);
```

Comment:
> CrateTrackSelectResult CrateStorage::selectCrateTracksSortedByCrateNameLike(const QString& crateNameLike) const

Ah, I see. By returning a CrateTrackSelectResult, CrateStorage can take care of everything touching the DB so the CrateFilterNode only needs the const reference to CrateStorage instead of the reference to the QSqlDatabase.